Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift

```diff
@@ -69,66 +69,68 @@ final class ApproveEngine {
-            throw Error.relayNotFound
+            throw Errors.relayNotFound
         }
 
         let proposeResponse = SessionType.ProposeResponse(relay: relay, responderPublicKey: selfPublicKey.hexRepresentation)
         let response = JSONRPCResponse<AnyCodable>(id: payload.wcRequest.id, result: AnyCodable(proposeResponse))
-        networkingInteractor.respond(topic: payload.topic, response: .response(response)) { _ in }
-
-        try settle(topic: sessionTopic, proposal: proposal, namespaces: sessionNamespaces)
+        
+        try await networkingInteractor.respond(topic: payload.topic, response: .response(response))
+        try await settle(topic: sessionTopic, proposal: proposal, namespaces: sessionNamespaces)
     }
     
-    func reject(proposerPubKey: String, reason: ReasonCode) throws {
+    func reject(proposerPubKey: String, reason: ReasonCode) async throws {
         guard let payload = try proposalPayloadsStore.get(key: proposerPubKey) else {
-            throw Error.proposalPayloadsNotFound
+            throw Errors.proposalPayloadsNotFound
         }
         proposalPayloadsStore.delete(forKey: proposerPubKey)
-        networkingInteractor.respondError(for: payload, reason: reason)
+        try await networkingInteractor.respondError(payload: payload, reason: reason)
         // TODO: Delete pairing if inactive
     }
     
-    func settle(topic: String, proposal: SessionProposal, namespaces: [String: SessionNamespace]) throws {
+    func settle(topic: String, proposal: SessionProposal, namespaces: [String: SessionNamespace]) async throws {
         guard let agreementKeys = try kms.getAgreementSecret(for: topic) else {
-            throw Error.agreementMissingOrInvalid
+            throw Errors.agreementMissingOrInvalid
         }
         let selfParticipant = Participant(
             publicKey: agreementKeys.publicKey.hexRepresentation,
             metadata: metadata
         )
-        let expectedExpiryTimeStamp = Date().addingTimeInterval(TimeInterval(WCSession.defaultTimeToLive))
-        
         guard let relay = proposal.relays.first else {
-            throw Error.relayNotFound
+            throw Errors.relayNotFound
         }
 
         // TODO: Test expiration times
+        let expiry = Date()
+            .addingTimeInterval(TimeInterval(WCSession.defaultTimeToLive))
+            .timeIntervalSince1970
+    
         let settleParams = SessionType.SettleParams(
             relay: relay,
             controller: selfParticipant,
             namespaces: namespaces,
-            expiry: Int64(expectedExpiryTimeStamp.timeIntervalSince1970)
-        )
+            expiry: Int64(expiry))
+
         let session = WCSession(
             topic: topic,
             selfParticipant: selfParticipant,
             peerParticipant: proposal.proposer,
             settleParams: settleParams,
-            acknowledged: false
-        )
+            acknowledged: false)
```

Comment:
I always ask myself whether to break a line or not when closing a parentheses 😅

Reply:
I almost always wrap to the next line. Don't remember why I decided to do this 😅